app/controllers/admin/users_controller.rb

```diff
@@ -60,7 +60,11 @@ def update
       Credits::Manage.call(@user, credit_params)
       add_note if user_params[:new_note]
 
-      redirect_to admin_user_path(params[:id])
+      if request.path == admin_users_path
```

Comment:
Hey @juliannatetreault  👋 

Soooo I made a [loom video](https://www.loom.com/share/8f4324cb8ff7404ab9a15f38817e280b) to do some debugging and walk you through my thought process.

However the _TLDR_ is that we when we assign a role we're actually calling a data form action as below

```
                    <button class="c-btn align-left w-100 c-btn--icon-left"
                    data-modal-title="Add role"
                    data-modal-size="small"
                    data-modal-content-selector="#add-role-modal"
                    data-form-action="<%= user_status_admin_user_path(user) %>">
                    <%= crayons_icon_tag("badge", aria_hidden: true, class: "c-btn__icon") %>
                    Assign role
                    </button>
```

hence, this form action points to the `user_status` action in the `users_controller` and not the `update` action where you've added the `if` statement above. 

However, when we move the code to the `user_status` action, you'll notice that it still doesnt obtain the desired results because the request path will be the `user_status_admin_user_path` and not just the `admin_users_path`. 

Hence, in this case we may need to use the referrer (`request.referrer`) to determine the address of the web page from which the resource has been requested. 

Please ping me if you have any questions! 😊 Hope this helps!

Reply:
Oh my goodness... 🤦‍♀️ I was using `request.referer`, just in the wrong spot! Thank you for rubber-ducking with me, @Ridhwana.